Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -100,6 +100,22 @@ public enum VersionSetSpecifier: Equatable, CustomStringConvertible {
         }
     }
 
+    public var hashValue: Int {
+        switch (self) {
+        case .any:
+            return 0x1
+
+        case .empty:
+            return 0x2
+
+        case .range(let range):
+            return 0x4 ^ range.lowerBound.hashValue ^ range.upperBound.hashValue
```

Comment:
I would include a multiply of (range.lowerBound.hashValue &* 31), to try and make these bits not cancel each other.